### DISCUSSION TITLE
Ausgabe aller aufgenommenen Audioelemente

### DISCUSSION THREAD

**Discussion by OmerAdam34** [neuefische/questions#159] [Web] [answered]
Hi, 
ich habe folgendes Problem: Mein Audioelement, nach dem aufgenommen wurde, soll unter "MY RECORDED TRACKS" ausgegeben werden. So das der User die Möglichkeit hat, seine aufnahmen zu sammeln. 
Daher habe ich versucht über das Audioelement zu Mappen und diese ebenfalls zu filtern. Ich habe mehrere Varianten ausprobiert und mich mit meinen Kommilitonen ausgetauscht. Leider komme ich diesbezüglich zu keinem Ergebnis. 

https://github.com/OmerAdam34/capstone-project/pull/13#issue-1382252099

**Comment by mbosselmann** [comment, marked as answer]
Hallo Omer,
dein `audiofile`, dass du in deiner Funktion `stopRecording` erstellst, wird nicht in deinem state `addRecordings` gespeichert, was du gut sehen kannst, wenn du dir den state ausloggst. 
Ich würde die Funktion `handleAddRecording` nicht bei einem event aufrufen, sondern in der Funktion `stopRecording` wie die anderen states, die du dort updatest. Hierfür würde ich das erstellte `file` an diese Funktion übergeben und für das neue Objekt, dass du dann hinzufügst, statt `audioPlayer` das `file` als Wert setzen. Hilft dir das schon weiter?

**Reply by OmerAdam34** [reply]
Hi Mareike, vielen Dank für deine Rückmeldung.  Also. ich habs versucht aber leider komme ich immer noch nicht ans gewünschte Ergebnis :/

**Reply by OmerAdam34** [reply]
Klasse! Mit Hilfe von den Coaches Mareike als auch Thomas, konnten wir das Problem beheben. Einige States wurden aus dem Code entnommen, da diese durch die Änderungen überflüssig waren. 

Der Fokus war in der Funktion handleAddRecording. Hier mussten zwei weitere und wichtige Parameter hinzugefügt werden, die wiederum im spread operator(welche wichtig für die spätere Ausgabe des objects ist)  wieder zu finden waren (blobUrl, audiofile). --> HINT! Einen Key prop für jedes der child Elemente muss ebenfalls gegeben sein. Daher auch die Vergabe von nanoID.
